From 8a115584261b32308d604063b56f25330ce8adaf Mon Sep 17 00:00:00 2001
From: Andrea Claudi <aclaudi@redhat.com>
Date: Mon, 29 Apr 2019 20:08:07 +0200
Subject: [PATCH] tc/q_multiq: Don't pass garbage in TCA_OPTIONS

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1465646
Upstream Status: iproute2.git commit 82ed9ffa2bb86

commit 82ed9ffa2bb86eea653f68a0ade945b7708818c9
Author: Phil Sutter <phil@nwl.cc>
Date:   Mon Aug 21 11:27:04 2017 +0200

    tc/q_multiq: Don't pass garbage in TCA_OPTIONS

    multiq_parse_opt() doesn't change 'opt' at all. So at least make sure
    it doesn't fill TCA_OPTIONS attribute with garbage from stack.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 tc/q_multiq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tc/q_multiq.c b/tc/q_multiq.c
index 7823931494563..9c09c9a7748f6 100644
--- a/tc/q_multiq.c
+++ b/tc/q_multiq.c
@@ -43,7 +43,7 @@ static void explain(void)
 static int multiq_parse_opt(struct qdisc_util *qu, int argc, char **argv,
 			    struct nlmsghdr *n)
 {
-	struct tc_multiq_qopt opt;
+	struct tc_multiq_qopt opt = {};
 
 	if (argc) {
 		if (strcmp(*argv, "help") == 0) {
-- 
2.20.1

